src/sentry_plugins/github/endpoints/webhook.py

```diff
@@ -273,9 +273,87 @@ def __call__(self, event, organization=None):
             self._handle(event, org, is_apps)
 
 
+class PullRequestEventWebhook(Webhook):
+    # https://developer.github.com/v3/activity/events/types/#pullrequestevent
+    def __call__(self, event, organization):
+        action = event['action']
+        is_apps = 'installation' in event
+        try:
+            repo = Repository.objects.get(
+                organization_id=organization.id,
+                provider='github_apps' if is_apps else 'github',
+                external_id=six.text_type(event['repository']['id']),
+            )
+
+        except Repository.DoesNotExist:
+            raise Http404()
+
+        # We need to track GitHub's "full_name" which is the repository slug.
+        # This is needed to access the API since `external_id` isn't sufficient.
+        if repo.config.get('name') != event['repository']['full_name']:
+            repo.config['name'] = event['repository']['full_name']
+            repo.save()
+
+        if action == 'opened':
+            self._handle_created(event, organization, repo, is_apps)
+        if action == 'edited' or action == "closed":
+            self._handle_updated(event, organization, repo)
+
+    def _handle_created(self, event, organization, repo, is_apps):
+        """PR was created"""
+
+        pull_request = event['pull_request']
+        number = pull_request['number']
+        title = pull_request['title']
+        body = pull_request['body']
+        user = pull_request['user']
+        merge_commit_sha = pull_request['merge_commit_sha']
+
+        # get author (TODO: use more complicated logic from the commit webhook)
+        author = author = CommitAuthor.objects.get_or_create(
+            organization_id=organization.id,
+            external_id=user['id'],
+            defaults={
+                'name': user['login']
+            }
+        )[0]
+
+        PullRequest.objects.create(
+            repository_id=repo.id,
+            organization_id=organization.id,
+            key=number,
+            title=title,
+            message=body,
+            author=author,
+            merge_commit_sha=merge_commit_sha,
+        )
+
+    def _handle_updated(self, event, organization, repo):
+        """PR title or description was edited, or PR was closed"""
+
+        pull_request = event['pull_request']
+        number = pull_request['number']
+        title = pull_request['title']
+        body = pull_request['body']
+        merge_commit_sha = pull_request['merge_commit_sha']
+
+        try:
+            PullRequest.objects.get(
+                repository_id=repo.id,
+                key=number,
+            ).update(
+                title=title,
+                message=body,
+                merge_commit_sha=merge_commit_sha,
+            )
+        except PullRequest.DoesNotExist:
+            pass
```

Comment:
if you're not going to do anything as a result of the `PullRequest.DoesNotExist` i'm pretty sure you could make this a single query instead of two by using `filter` instead of `get`